include/deal.II/base/symmetric_tensor.h

```diff
@@ -915,6 +915,17 @@ namespace internal
 
 template <int rank, int dim, typename Number>
 inline
+SymmetricTensor<rank,dim,Number>::SymmetricTensor ()
+{
+  // Some auto-differentiable numbers need explicit
+  // zero initialization.
+  for (unsigned int i=0; i<base_tensor_type::dimension; ++i)
+    data[i] = internal::NumberType<Number>::value(0.0);
```

Comment:
Hmm... true. Good catch! I'll write some test cases as you suggested and, if there is a problem, implement an alternate strategy for initialisation.